readme.md

```diff
@@ -45,6 +45,21 @@ Because this proposal allows existing contracts to remove higher-precision math
 ## Technical Specification
 
 The limit on the maximum length of Bitcoin Cash VM numbers (A.K.A. `nMaxNumSize`) is removed.
+The numbers will still be limited, but by the maximum stack item size (A.K.A. `MAX_SCRIPT_ELEMENT_SIZE`).
+The effective limits will then be a calculated value from `MAX_SCRIPT_ELEMENT_SIZE`:
+
+- `MAX_SCRIPTNUM = (MAX_SCRIPT_ELEMENT_SIZE * 8 - 1) - 1`, and
+- `MIN_SCRIPTNUM = -MAX_SCRIPTNUM`,
+
+rather than independently set limits (as they were before this upgrade).
+
+The [`CHIP: Targeted Virtual Machine Limits`](https://github.com/bitjson/bch-vm-limits) will set `MAX_SCRIPT_ELEMENT_SIZE` to 10,000 bytes, therefore valid range for results of arithmetic operations shall be the inclusive range: `[-2^79999 + 1, 2^79999 - 1]`.
+
+Any operation that would result in a stack item bigger than `MAX_SCRIPT_ELEMENT_SIZE` must fail, e.g. `{stack: a} OP_1ADD OP_0NOTEQUAL` must fail for `a = MAX_SCRIPTNUM`.
+
+All implementations must fail such script and should throw an error, but it is their choice whether to throw a generic error for operation exceeding `MAX_SCRIPT_ELEMENT_SIZE` or a specific error like `ScriptError::INVALID_NUMBER_RANGE_BIG_INT`.
+
+By coupling the numerical limit with `MAX_SCRIPT_ELEMENT_SIZE` we can guarantee that **any stack item** can be a valid input to an arithmetic operation, e.g. `{stack: a} OP_BIN2NUM OP_ABS OP_0NOTEQUAL` will pass for any `a` on stack, since it will not be possible to have a stack item that would be interpreted as value outside the `[-2^79999 + 1, 2^79999 - 1]` range.
```

Comment:
I'd just leave this off. It's mixing the implementation detail too much into the spec.

Reply:
Alternatively use the root metric of byte size to say the same thing.

Reply:
how's this: https://github.com/bitjson/bch-bigint/pull/8/commits/2105310be8f6af9e91a0a902b46cb6e34dba8661

Reply:
also nit: fixed the script to pass for value 0, too

https://github.com/bitjson/bch-bigint/pull/8/commits/d39bba94f0ee6a0f2ad073ed1b311a29510995f6